Implements `try_state` hook in elections and EPM pallets

gpestana opened the pull request:
This PR implements the `try_state` hook in elections and EPM pallets and uses it in the tests. An improvement to this PR would be to work on https://github.com/paritytech/polkadot-sdk/issues/210 to automatically include calling the `try_state` hook in all tests. 

Related to https://github.com/paritytech/polkadot-sdk/issues/239

yjhmelody committed:
chore(sc-cli): improve runner and signals (#13688)

* chore(sc-cli): improve runner and signals

* Update client/cli/src/runner.rs

* fmt

---------

Co-authored-by: Bastian Köcher <git@kchr.de>

Commit:
add a new ci job to fuzz sp-arithmetic (#13673)

as requested in https://github.com/paritytech/ci_cd/issues/761

bkchr committed:
sc-slots: Forward block size limit (#13716)

gpestana committed:
Adds try_state to elections pallet

gavofyork committed:
FRAME: Fix the Referenda confirming alarm (#13704)

* Fix the Referenda confirming alarm

* Add minimal regression test

This fails on bf395c8308c481a9774373e0b0b14bd7a2e4b8d2 since the
downwards rounding voids the curve delay.

Signed-off-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>

---------

Signed-off-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>
Co-authored-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>

aaronbassett committed:
Update pallet licenses to Apache-2.0 (#13467)

Commit:
Reward pool migration fix (#13715)

* reward pool migration fix

* comment

* remove generic

* rm max

* foramtting

* Add note to V4 migration

Signed-off-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>

* Add more asserts to the V5 migration

Signed-off-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>

* Make compile

Signed-off-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>

* Update frame/nomination-pools/src/migration.rs

Co-authored-by: Gonçalo Pestana <g6pestana@gmail.com>

* Make V4 migration re-usable

Otherwise it wont chain together with the V5.

Signed-off-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>

* Add MigrateV3ToV5 wrapper

Signed-off-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>

---------

Signed-off-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>
Co-authored-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>
Co-authored-by: Gonçalo Pestana <g6pestana@gmail.com>

pgherveou committed:
contracts: Add host function tracing (#13648)

aaronbassett committed:
Change license of node-template and FRAME examples to MIT-0 (#13465)

* Change node-template license from Unlicense to MIT-0

* Change frame examples license from Unlicense to MIT-0

* Update bin/node-template/LICENSE

---------

Co-authored-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>
Co-authored-by: parity-processbot <>

the-right-joyce committed:
updating labels descriptions (#13557)

* updating labels descriptions

* delete milestones

* Update docs/CONTRIBUTING.adoc

Co-authored-by: Bastian Köcher <git@kchr.de>

* link to label docs

* Update docs/CONTRIBUTING.adoc

Co-authored-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>

---------

Co-authored-by: Bastian Köcher <git@kchr.de>
Co-authored-by: parity-processbot <>
Co-authored-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>

rcny committed:
CI: Investigate why `RUSTFLAGS` differs on `master` and PRs (#13686)

skunert committed:
Remove HeaderBackend requirement from AuthorityDiscovery and NetworkWorker (#13730)

* Remove `HeaderBackend` requirement from `NetworkWorker`

* Remove HeaderBackend from authority-discovery

rcny committed:
CI: rephrase `RUSTFLAGS` (#13735)

kziemianek committed:
contracts: proper event link in docs (#13729)

gpestana committed:
Finishes elections try-state hook; starts EPM hook impl

gpestana committed:
Adds scaffolding for EPM try_state checks

paritytech-cicd-pr commented:
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2604560

bkchr committed:
Support stable rust for compiling the runtime (#13580)

* Support stable rust for compiling the runtime

This pull request brings support for compiling the runtime with stable Rust. This requires at least
rust 1.68.0 to work on stable. The code is written in a way that it is backwards compatible and
should automatically work when someone compiles with 1.68.0+ stable.

* We always support nightlies!

* :facepalm:

* Sort by version

* Review feedback

* Review feedback

* Fix version parsing

* Apply suggestions from code review

Co-authored-by: Koute <koute@users.noreply.github.com>

---------

Co-authored-by: Koute <koute@users.noreply.github.com>

Commit:
provide a default value for RELENG_SCRIPTS_BRANCH (#13743)

Currently this variable is configured exclusively via a project
variable, which makes it impossible to override for specific test prs.

Commit:
roll out new debian 11 ci image (#13744)

davxy committed:
Swap 'base58' with 'bs58' (#13739)

* Swap base58 with bs58

* Removed unused clone

* std flag

michalkucharczyk committed:
proc-macro: check for non-args runtime calls added (#13742)

* proc-macro: check for non-args runtime calls added

* Update primitives/api/proc-macro/src/impl_runtime_apis.rs

Co-authored-by: Bastian Köcher <git@kchr.de>

---------

Co-authored-by: Bastian Köcher <git@kchr.de>
Co-authored-by: parity-processbot <>

Commit:
Clean up after debian 11 rollout (#13762)

This reverts commit 56d8cbfcac926439aa78b684b7c77921dd2f4a7f.

ruseinov committed:
[Fix] Bump tuple element number in frame-support. (#13760)

That is to avoid hitting the pallet limit that emits a cryptic error. More detail could be found here: https://substrate.stackexchange.com/questions/7212 .

Co-authored-by: parity-processbot <>

davxy committed:
Generic keystore internals (#13749)

* Generic testing keystore internals

* Generic local keystore internals

* Restore deleted comment

davxy committed:
Application Crypto cleanup (#13746)

* Adjust application crypto docs

* Blanket implementation for 'RuntimeAppPublic' trait

* Blanket implementation for 'BoundToRuntimeAppPublic' for 'RuntimeAppPublic'

* Relax type bounds

* Docs fix

* restore MaybeHash

* Commit suggestion

Co-authored-by: Anton <anton.kalyaev@gmail.com>

---------

Co-authored-by: Anton <anton.kalyaev@gmail.com>

altonen committed:
Attempt to relieve pressure on `mpsc_network_worker` (#13725)

* Attempt to relieve pressure on `mpsc_network_worker`

`SyncingEngine` interacting with `NetworkWorker` can put a lot of strain
on the channel if the number of inbound connections is high. This is
because `SyncingEngine` is notified of each inbound substream which it
then can either accept or reject and this causes a lot of message
exchange on the already busy channel.

Use a direct channel pair between `Protocol` and `SyncingEngine`
to exchange notification events. It is a temporary change to alleviate
the problems caused by syncing being an independent protocol and the
fix will be removed once `NotificationService` is implemented.

* Apply review comments

* fixes

* trigger ci

* Fix tests

Verify that both peers have a connection now that the validation goes
through `SyncingEngine`. Depending on how the tasks are scheduled,
one of them might not have the peer registered in `SyncingEngine` at which
point the test won't make any progress because block announcement received
from an unknown peer is discarded.

Move polling of `ChainSync` at the end of the function so that if a block
announcement causes a block request to be sent, that can be sent in the
same call to `SyncingEngine::poll()`.

---------

Co-authored-by: parity-processbot <>

ruseinov committed:
[Enhancement] Throw an error when there are too many pallets (#13763)

* [Enhancement] Throw an error when there are too many pallets

* fix ui test

* fix PR comments

* Update frame/support/procedural/src/construct_runtime/mod.rs

Co-authored-by: Bastian Köcher <git@kchr.de>

* Update frame/support/procedural/src/construct_runtime/mod.rs

Co-authored-by: Bastian Köcher <git@kchr.de>

* ".git/.scripts/commands/fmt/fmt.sh"

---------

Co-authored-by: Bastian Köcher <git@kchr.de>
Co-authored-by: command-bot <>

athei committed:
Build wasm for mvp cpu (#13758)

acatangiu committed:
BEEFY: gossip finality proofs (#13727)

* sc-consensus-beefy: add justifications to gossip protocol

* sc-consensus-beefy: voter gossips finality proofs

* sc-consensus-beefy: add finality proof gossip test

* sc-consensus-beefy: always gossip finality proof

Gossip finality proof in _both_ cases of reaching finality threshold
through votes:
1. threshold reached through self vote,
2. threshold reached through incoming vote.

* address comments

kianenigma committed:
Fix nomiantion pools doc render (#13748)

Co-authored-by: parity-processbot <>

gpestana committed:
implements checks for EPM

bkchr committed:
sp-runtime-interface-test: Fix flaky test (#13770)

pepoviola committed:
bump zombienet version (#13772)

pgherveou committed:
[Contracts] Overflowing bounded `DeletionQueue` allows DoS against contract termination (#13702)

* [Contracts review] Overflowing bounded `DeletionQueue` allows DoS against contract termination

* wip

* wip

* wip

* wip

* wip

* fix doc

* wip

* PR review

* unbreak tests

* fixes

* update budget computation

* PR comment: use BlockWeights::get().max_block

* PR comment: Update queue_trie_for_deletion signature

* PR comment: update deletion budget docstring

* PR comment: impl Default with derive(DefaultNoBound)

* PR comment: Remove DeletedContract

* PR comment Add ring_buffer test

* remove missed comment

* misc comments

* contracts: add sr25519_recover

* Revert "contracts: add sr25519_recover"

This reverts commit d4600e00934b90e5882cf5288f36f98911b51722.

* ".git/.scripts/commands/bench/bench.sh" pallet dev pallet_contracts

* PR comments update print_schedule

* Update frame/contracts/src/benchmarking/mod.rs

* Update frame/contracts/src/storage.rs

* Update frame/contracts/src/storage.rs

* rm temporary fixes

* fix extra ;

* Update frame/contracts/src/storage.rs

Co-authored-by: juangirini <juangirini@gmail.com>

* Update frame/contracts/src/storage.rs

Co-authored-by: Alexander Theißen <alex.theissen@me.com>

* Update frame/contracts/src/lib.rs

Co-authored-by: Alexander Theißen <alex.theissen@me.com>

* Update frame/contracts/src/lib.rs

Co-authored-by: Alexander Theißen <alex.theissen@me.com>

* Support stable rust for compiling the runtime (#13580)

* Support stable rust for compiling the runtime

This pull request brings support for compiling the runtime with stable Rust. This requires at least
rust 1.68.0 to work on stable. The code is written in a way that it is backwards compatible and
should automatically work when someone compiles with 1.68.0+ stable.

* We always support nightlies!

* :facepalm:

* Sort by version

* Review feedback

* Review feedback

* Fix version parsing

* Apply suggestions from code review

Co-authored-by: Koute <koute@users.noreply.github.com>

---------

Co-authored-by: Koute <koute@users.noreply.github.com>

* github PR commit fixes

* Revert "Support stable rust for compiling the runtime (#13580)"

This reverts commit 0b985aa5ad114a42003519b712d25a6acc40b0ad.

* Restore DeletionQueueMap

* fix namings

* PR comment

* move comments

* Update frame/contracts/src/storage.rs

* Update frame/contracts/src/storage.rs

* fixes

---------

Co-authored-by: command-bot <>
Co-authored-by: juangirini <juangirini@gmail.com>
Co-authored-by: Alexander Theißen <alex.theissen@me.com>
Co-authored-by: Bastian Köcher <git@kchr.de>
Co-authored-by: Koute <koute@users.noreply.github.com>

rphmeier committed:
Refactor: extract most aura logic out to standalone module, make use of these (#13764)

* Extract most aura logic out to standalone module, make use of these

* Update client/consensus/aura/src/standalone.rs

improve docs

Co-authored-by: Bastian Köcher <git@kchr.de>

* add slot_duration_at

* ".git/.scripts/commands/fmt/fmt.sh"

---------

Co-authored-by: Bastian Köcher <git@kchr.de>
Co-authored-by: parity-processbot <>

pgherveou committed:
contracts: make test work with debugger (#13776)

* contracts: make test work with debugger

* fix path

* PR review

* Add comment

* space

* Update frame/contracts/src/tests.rs

* lint

* spelling

Commit:
add claim_commission weight (#13774)

gavofyork committed:
FRAME: Minor fix for failsafe. (#13741)

* max instead of min

* Remove debug stuff

* remove debug code

* warn log on no provider ref

* format string for log

---------

Co-authored-by: muharem <ismailov.m.h@gmail.com>

DamianStraszak committed:
Adjustments to RPC-query docstrings. (#13698)

* Changes in RPC docs.

* ".git/.scripts/commands/fmt/fmt.sh"

---------

Co-authored-by: command-bot <>

bkchr committed:
Force upgrade snow to 0.9.2 (#13806)

This fixes the compilation on master for the node template that was not pulling the latest release
as part of its build.

muharem committed:
Scheduler pre block limit note (#13231)

Co-authored-by: parity-processbot <>

koute committed:
Disable `sign-ext` WASM feature when building runtimes (#13804)

Co-authored-by: parity-processbot <>

wischli committed:
refactor: inconsistent BalanceConversion fn (#13610)

* refactor: inconsistent BalanceConversion fn

* Revert "refactor: inconsistent BalanceConversion fn"

This reverts commit 1177877631a11b64df6f019b5390a8a7018e3a3f.

* refactor: rename BalanceConversion trait

* feat: add ConversionFromAssetBalance

bkchr committed:
Remove deprecated batch verification (#13799)

This removes the deprecated batch verification. This was actually never really activated.
Nevertheless, we need to keep the host functions around to support old runtimes which may import
these host functions. However, we do not give access to these functions anymore. This means that any new
runtime can not call them anymore. The host function implementations we keep will not do batch verification and will
instead fall back to the always existing option of directly verifying the passed signature.
`finish_batch_verification` will return the combined result of all the batch verify calls.

This removes the `TaskExecutorExt` which only existed to support the batch verification. So, any
code that used this extension can just remove the registration of them. It also removes
`SignatureBatching` that was used by `frame-executive` to control the batch verification.
However, there wasn't any `Verify` implementation that called the batch verification functions.

ggwpez committed:
Uniform pallet warnings (#13798)

* Use proc-macro-warning crate

Signed-off-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>

* Fixup

Signed-off-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>

* Fix pallet_ui tests

Also renamed some of the odd-named ones.

Signed-off-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>

* Update dep

Signed-off-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>

* Ignore hardcoded weight warning

To be fixed in https://github.com/paritytech/substrate/issues/13813

Signed-off-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>

* Fix test pallet

Signed-off-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>

* Fix more tests

Signed-off-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>

---------

Signed-off-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>
Co-authored-by: parity-processbot <>

simonsso committed:
Update documentation for uniques (This PR renames classes and instances to collections and items in the Uniques pallet in order to follow the commonly accepted NFTs terminology.) (#13322)

* Update documentation for uniques

The documentation was outdated after merge of #11389
Using the widely spread term collections and item instead
of the previous class and instance.

* Update README.md

---------

Co-authored-by: Jegor Sidorenko <5252494+jsidorenko@users.noreply.github.com>
Co-authored-by: Jegor Sidorenko <jegor@parity.io>

KiChjang committed:
Implement #[pallet::composite_enum] (#13722)

* Implement #[pallet::hold_reason]

* Appease clippy

* cargo fmt

* Update test expectations

* Update test expectations

* Support composite_enum attribute instead

* Update test expectations

* Change hold_reason to composite_enum

* Add UI test for unsupported identifier when using composite_enum

* Fix comment

* Add documentation for pallet::composable_enum

* More docs

* cargo fmt

s0me0ne-unkn0wn committed:
Expose WASM extensions in executor semantics (#13811)

* Expose WASM extensions in executor semantics

* Fix benches

* Remove redundant extensions

ggwpez committed:
Deprecate V1 Weights (#13699)

* Remove deprecated pallet calls

Signed-off-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>

* Deprecate old weight

Signed-off-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>

* Update Runtime API

Signed-off-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>

* Fix tests

Signed-off-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>

* Delete shitty code

Signed-off-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>

* Fix doctest

Signed-off-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>

* Update frame/alliance/src/lib.rs

Co-authored-by: Koute <koute@users.noreply.github.com>

* Add doc

Signed-off-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>

* contracts: Use u64 as old weight type

Signed-off-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>

* Update frame/contracts/src/lib.rs

Co-authored-by: Alexander Theißen <alex.theissen@me.com>

---------

Signed-off-by: Oliver Tale-Yazdi <oliver.tale-yazdi@parity.io>
Co-authored-by: Koute <koute@users.noreply.github.com>
Co-authored-by: Alexander Theißen <alex.theissen@me.com>
Co-authored-by: parity-processbot <>

agryaznov committed:
update links to ink! (#13819)

bkchr committed:
Move registration of `ReadRuntimeVersionExt` to `ExecutionExtension` (#13820)

Instead of registering `ReadRuntimeVersionExt` in `sp-state-machine` it is moved to
`ExecutionExtension` which provides the default extensions.

gpestana committed:
Mention `on_idle` round-robin logic to `trait Hooks` cargo doc (#13797)

* Adds on_idle round-robin logic to trait Hooks docs

* Makes the docs more concise

* Update frame/support/src/traits/hooks.rs

Co-authored-by: Bastian Köcher <git@kchr.de>

* fmt

---------

Co-authored-by: Bastian Köcher <git@kchr.de>

bkchr committed:
ProofRecorder: Implement transactional support (#13769)

* TrieRecorder: Start adding support for transactions

* Adds `transactions` functions and some test

* More tests

* Docs

* Ensure that we rollback failed transactions in the storage proof

* FMT

* Update primitives/trie/src/recorder.rs

Co-authored-by: Dmitry Markin <dmitry@markin.tech>

* Review comments

* Update primitives/trie/src/recorder.rs

Co-authored-by: Sebastian Kunert <skunert49@gmail.com>

* ".git/.scripts/commands/fmt/fmt.sh"

* For the holy clippy!

* Update primitives/trie/src/recorder.rs

Co-authored-by: Anton <anton.kalyaev@gmail.com>

---------

Co-authored-by: Dmitry Markin <dmitry@markin.tech>
Co-authored-by: Sebastian Kunert <skunert49@gmail.com>
Co-authored-by: command-bot <>
Co-authored-by: Anton <anton.kalyaev@gmail.com>

liamaharon committed:
remove duplicate sync option documentation (#13828)

KiChjang committed:
Add HoldReason to the NIS pallet (#13823)

* Add HoldReason to the NIS pallet

* Rename composable_enum to composite_enum

* Add encoding test

* Add more doc comments

liamaharon committed:
Fix `try-runtime follow-chain`, try-runtime upgrade tuple tests, cli test utils (#13794)

* new test for try-runtime tuple stuff

* fix

* remove development comment

* formatting

* remove todo comment

* follow-chain working test

* refactor common cli testing utils

* fix comment

* revert Cargo.lock changes

* update Cargo.lock

* improve doc comment

* fix error typo

* update Cargo.lock

* feature gate try-runtime test

* build_substrate cli test util

* feature gate follow_chain tests

* move fn start_node to test-utils

* improve test pkg name

* use tokio Child and Command

* remove redundant import

* fix ci

* fix ci

* don't leave hanging processes

* improved child process cleanup

* use existing KillChildOnDrop

* remove redundant comment

* Update test-utils/cli/src/lib.rs

Co-authored-by: Koute <koute@users.noreply.github.com>

---------

Co-authored-by: kianenigma <kian@parity.io>
Co-authored-by: Koute <koute@users.noreply.github.com>

alexbcasas committed:
Use proper weights in the `pallet-template` (#13817)

* benchmarking to generate weights file

* add the calculated weights in the extrinsics

* use benchmarking v2 syntax to generate the weights

* minor syntax change when benchmarking

* added WeightInfo in the mock to pass tests

* minor cargo fmt format changes

gpestana committed:
Update frame/election-provider-multi-phase/src/lib.rs

Co-authored-by: Liam Aharon <liam.aharon@hotmail.com>

gpestana committed:
Update frame/elections-phragmen/src/lib.rs

Co-authored-by: Liam Aharon <liam.aharon@hotmail.com>

paritytech-cicd-pr commented:
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2709994

gpestana commented on frame/election-provider-multi-phase/src/lib.rs:
```diff
@@ -1252,6 +1257,8 @@ pub mod pallet {
 	pub type CurrentPhase<T: Config> = StorageValue<_, Phase<T::BlockNumber>, ValueQuery>;
 
 	/// Current best solution, signed or unsigned, queued to be returned upon `elect`.
+	///
+	/// Invariant: Always sorted by score.
```
I'm removing `Invariant:` from this line. The other storage items do mention the invariant without the prefix.

gpestana commented on frame/election-provider-multi-phase/src/lib.rs:
```diff
@@ -1570,6 +1577,79 @@ impl<T: Config> Pallet<T> {
+
+	// [`SignedSubmissionsMap`] state check. Invariants:
+	// - All [`SignedSubmissionIndices`] are present in [`SignedSubmissionsMap`], and no more;
+	// - [`SignedSubmissionNextIndex`] is not present in [`SignedSubmissionsMap`];
```
>  Should `SignedSubmissionNextIndex` be equal to the size of `SignedSubmissionsMap`?
Not necessarily, that is the reason why we keep the index in storage (also to avoid iterating over the map).

gpestana commented on frame/elections-phragmen/src/lib.rs:
```diff
@@ -1192,6 +1197,96 @@ impl<T: Config> ContainsLengthBound for Pallet<T> {
+		}
+	}
+
+	// [`RunnersUp`] state checks. Invariants:
+	//  - Elements are sorted based on rank.
+	fn try_state_runners_up() -> Result<(), &'static str> {
+		Ok(())
+	}
```
Added!

gpestana committed:
Addresses PR review comments

gpestana commented on frame/election-provider-multi-phase/src/lib.rs:
```diff
@@ -1570,6 +1577,79 @@ impl<T: Config> Pallet<T> {
+		Self::try_state_phase_off()
+	}
+
+	// [`Snapshot`] state check. Invariants:
+	// - [`DesiredTargets`] exist IFF [`Snapshot`] is present.
+	// - [`SnapshotMetadata`] exist IFF [`Snapshot`] is present.
```
Added one more check to ensure `DesiredTargets` and `SnapshotMetadata` are not set if `Snapshot` is not set.

paritytech-cicd-pr commented:
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2718017

gpestana committed:
Merge branch 'master' into gpestana/try_state_elections

paritytech-cicd-pr commented:
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-check-benches
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2718018

gpestana commented:
bot rebase

gpestana commented:
Closing in lieu of https://github.com/paritytech/substrate/pull/13979
